src/main/java/no/unit/nva/cristin/projects/CristinApiClient.java

```diff
@@ -138,14 +141,64 @@ protected CristinProject getProject(String id, String language) throws ApiGatewa
         throws ApiGatewayException {
 
         List<CristinProject> projectsFromQuery = asList(getDeserializedResponse(response, CristinProject[].class));
+        List<URI> cristinUris = extractCristinUrisFromProjects(language, projectsFromQuery);
+        List<HttpResponse<String>> individualResponses = fetchQueryResultsOneByOne(cristinUris);
+        List<CristinProject> enrichedCristinProjects = mapValidResponsesToCristinProjects(individualResponses);
+        enrichedCristinProjects =
+            combineResultsWithQueryInCaseEnrichmentFails(projectsFromQuery, enrichedCristinProjects);
 
+        return enrichedCristinProjects;
+    }
+
+    protected List<CristinProject> combineResultsWithQueryInCaseEnrichmentFails(
+        List<CristinProject> projectsFromQuery,
+        List<CristinProject> enrichedProjects) {
+
+        if (projectsFromQuery.size() == enrichedProjects.size()) {
+            return enrichedProjects;
+        }
+
+        List<CristinProject> missingProjects = projectsFromQuery.stream()
```

Comment:
If original project from `projectsFromQuery` has no match on `cristinProjectId` in `enrichedProjects`, it gets added to this list of missing projects